tasks/apply-mapping/apply-mapping.yaml

```diff
@@ -93,12 +92,72 @@ spec:
             echo -n $tags | jq -c
         }
 
-        # Merge the mapping key contents in the data json file with the components key in the snapshot based
-        # on component name. Save the output as a compact json in mapped_snapshot.json file in the workspace
-        { echo -n $(cat "${SNAPSHOT_SPEC_FILE_ORIG}"); echo "${MAPPING}"; } | jq -c -s '.[0] as $snapshot
-          | .[0].components + .[1].components | group_by(.name) | [.[] | select(length > 1)]
-          | map(reduce .[] as $x ({}; . * $x)) as $mergedComponents | $snapshot | .components = $mergedComponents' \
-          > "${SNAPSHOT_SPEC_FILE}"
+        convert_to_quay () { # Convert the registry.redhat.io URL to the quay.io format
+            local repository=$1
+            case "$repository" in
+                registry.redhat.io/*)
+                    echo "${repository/registry.redhat.io/quay.io/redhat-prod}" \
+                        | sed 's|/|----|g; s|quay.io----redhat-prod----|quay.io/redhat-prod/|'
+                    ;;
+                registry.stage.redhat.io/*)
+                    echo "${repository/registry.stage.redhat.io/quay.io/redhat-pending}" \
+                        | sed 's|/|----|g; s|quay.io----redhat-pending----|quay.io/redhat-pending/|'
+                    ;;
+                *)
+                    echo "$repository"
+                    ;;
+            esac
+        }
+
+        # This block is temporary to support both quay.io and registry.redhat.io
+        # It should be removed once all repositories are migrated to registry.redhat.io
+        convert_to_registry () { # Convert the repository URL to the registry.redhat.io format
+            local repository=$1
+            case "$repository" in
+                quay.io/redhat-prod/*)
+                    echo "${repository/quay.io\/redhat-prod/registry.redhat.io}" | sed 's|----|/|g'
+                    ;;
+                quay.io/redhat-pending/*)
+                    echo "${repository/quay.io\/redhat-pending/registry.stage.redhat.io}" | sed 's|----|/|g'
+                    ;;
+                registry.redhat.io/* | registry.stage.redhat.io/*)
+                    # Return the original Red Hat registry paths
+                    echo "$repository"
+                    ;;
+                *)
+                    # Return empty for unhandled formats
+                    echo ""
+                    ;;
+            esac
+        }
+
+        convert_to_registry_access () { # Convert the repository URL to the registry.access.redhat.com format
+            local repository=$1
+            case "$repository" in
+                quay.io/redhat-prod/*)
+                    echo "${repository/quay.io\/redhat-prod/registry.access.redhat.com}" | sed 's|----|/|g'
+                    ;;
+                quay.io/redhat-pending/*)
+                    echo "${repository/quay.io\/redhat-pending/registry.access.stage.redhat.com}" | sed 's|----|/|g'
+                    ;;
```

Comment:
These lines can be removed, because we will always convert from the rh_registry value.

Reply:
Fixed!